common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/microsoftsts/MicrosoftStsOAuth2Strategy.java

```diff
@@ -331,9 +357,33 @@ public MicrosoftStsTokenRequest createTokenRequest(@NonNull final MicrosoftStsAu
         tokenRequest.setRedirectUri(request.getRedirectUri());
         tokenRequest.setClientId(request.getClientId());
         tokenRequest.setScope(request.getTokenScope());
-
         tokenRequest.setGrantType(TokenRequest.GrantTypes.AUTHORIZATION_CODE);
+        setTokenRequestCorrelationId(tokenRequest);
+
+        try {
+            if (SCHEME_POP.equals(mStrategyOptions.getAuthenticationScheme().getName())) {
```

Comment:
TODO Add support for token **renewal**

Reply:
Resolving here, added TODO here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/733/files#r351052294